### PR TITLE
MOD-6890, MOD-7176: add expiring fields tests and handle hexpire notification

### PR DESCRIPTION
**Pull request by kei-nan**
A clear and concise description of what the PR is solving, including:
1. The current state briefly: We did not handle the newest hexpired notification, there were no tests for the field expiration feature
2. What is the change: Handling the hexpired notification and adding tests to test the field expiration feature
3. Adding the outcome: Better handling/monitoring expiring fields feature

**Which issues this PR fixes**
1. [MOD-6890](https://redislabs.atlassian.net/browse/MOD-6890): add expiring fields tests
2. [MOD-7176](https://redislabs.atlassian.net/browse/MOD-7176): support hexpire notification

**Main objects this PR modified**
1. Notifications: mainly hexpire
2. Tests: adding new tests

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

[MOD-6890]: https://redislabs.atlassian.net/browse/MOD-6890?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOD-7176]: https://redislabs.atlassian.net/browse/MOD-7176?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/notifications.c [hunk 1]
```diff
@@ -152,6 +154,7 @@ int HashNotificationCallback(RedisModuleCtx *ctx, int type, const char *event,
     case del_cmd:
     case set_cmd:
     case trimmed_cmd:
+    case hexpire_cmd:
```
Comment (oshadmi) [inline src/notifications.c:157]: Not sure this is the only update that needs to be done.
Key expiration is key deletion, but a field expiration could be either a key deletion (if last field expired) or only a field deletion so need to also look at the handling of `hdel_cmd`

File: tests/pytests/test_expire.py [hunk 2]
```diff
@@ -218,3 +220,89 @@ def expireDocs(env, isSortable, expected_results, isJson):
...
+    testExpireFields(env, field_name_to_expire={'x': None, 'y': 1},
+                     document_name_to_expire={'doc1': True, 'doc2': False},
+                     expected_results=[2, 'doc1', ['x', 'f'], 'doc2', ['x', 'f', 'y', 'f']])
+
+
+def testSingleSortableFieldWithExpiration(env):
+    testExpireFields(env, field_name_to_expire={'x': 1},
+                     document_name_to_expire={'doc1': True, 'doc2': False},
+                     field_to_schema_list={'x': ['SORTABLE']},
+                     expected_results=[2, 'doc1', ['x', 't'], 'doc2', ['x', 'f']])
```
Comment (raz-mon) [inline tests/pytests/test_expire.py:287]: How is this results returned? `doc1` should be deleted.
Please validate this test.

File: tests/pytests/test_expire.py [hunk 3]
```diff
@@ -218,3 +220,89 @@ def expireDocs(env, isSortable, expected_results, isJson):
...
+
+
+def testSortableFieldWithExpirationAndRegularField(env):
+    testExpireFields(env, field_name_to_expire={'x': 1, 'y': None},
+                     document_name_to_expire={'doc1': True, 'doc2': False},
+                     field_to_schema_list={'x': ['SORTABLE']},
+                     expected_results=[2, 'doc1', ['x', 't', 'y', 'f'], 'doc2', ['x', 'f', 'y', 'f']])
```
Comment (raz-mon) [inline tests/pytests/test_expire.py:294]: Same about `['x', 't']` - shouldn't be returned IIUC

**Review by raz-mon — COMMENTED** [review]
Nice!
See a few comments

File: tests/pytests/test_expire.py [hunk 4]
```diff
@@ -218,3 +221,150 @@ def expireDocs(env, isSortable, expected_results, isJson):
...
+    field_to_additional_schema_keywords = {'x': [], 'y': ['SORTABLE']}
+    schema = createTextualSchema(field_to_additional_schema_keywords)
+    commonFieldExpiration(env, schema, field_to_additional_schema_keywords.keys(),
+                          expiration_interval_to_fields={1: ['x'], 3: ['y']},
+                          document_name_to_expire={'doc1': True, 'doc2': False})
+
+
+# Aims to test that the expectation that for 2 fields with the same expiration time we will get a single notification
+@skip(redis_less_than='7.4')
+def testExpireMultipleFields(env):
```
Comment (raz-mon) [inline tests/pytests/test_expire.py:365]: Necessary?

File: tests/pytests/test_expire.py [hunk 5]
```diff
@@ -218,3 +221,150 @@ def expireDocs(env, isSortable, expected_results, isJson):
...
+
+def transform_document_list_to_dict(document_list):
+    if len(document_list) == 0:
+        return {}
+
+    result = {}
+    num_docs = document_list[0]
+    for i in range(1, num_docs * 2, 2):
+        result[document_list[i]] = {}
+        values_dict = result[document_list[i]]
```
Comment (raz-mon) [inline tests/pytests/test_expire.py:254]:
```suggestion
        values_dict = result[document_list[i]] = {}
```

**Review by raz-mon — COMMENTED** [review]
Looking good!
See few comments.

File: tests/pytests/test_expire.py (hunk 4 above)
Comment (kei-nan) [reply tests/pytests/test_expire.py:364]: Changed tests to be more unique

**Review by raz-mon — APPROVED** [review]

**Review by raz-mon — APPROVED** [review]

**Pull request merged**